index.html

```diff
@@ -14,29 +14,18 @@
 	<link href="css/achilles.css" rel="stylesheet">
 	<link href="css/chart.css" rel="stylesheet">
 
-	<script>
-		var report = 'dashboard';
-		var page_vm = {};
-
-		function updateReport(value) {
-			report = value;
-			updateRoute();
-		}
-
-		function setDatasource(index) {
-			page_vm.datasource(page_vm.datasources[index]);
-			updateRoute();
-		}
-
-		function updateRoute() {
-			$('.reportDrilldown').addClass('hidden');
-			document.location = '#/' + page_vm.datasource().folder + '/' + report;
-		}
-	</script>
-
 	<script src="js/curl.config.js"></script>
 	<script src="js/curl.js"></script>
+	<script src="js/jquery-1.10.2.js"></script>
 	<script src="js/app/achilles.js"></script>
+	<script src="js/app/achillesWrapper.js"></script>	
+	<script>
+		var datasourcepath = 'data/datasources.json'
+		var containerName = "#achillesContainer";
+		var achillespath = "achilles.html";
+		var a = new Achilles( datasourcepath, containerName, achillespath, achillesWeb);
```

Comment:
Yes, that's the idea. What sent me into this line of thining is that I don't usually expect to see in-line script (such as var a = new Achilles()).  The line with the new Achilles() call would go in the main app startup:

````javascript
curl([
	"jquery",
	"d3",
	"knockout",
	"common",
	"app/reports",
	"Achilles"
	"bootstrap",
	"d3/tip"
], function ($, d3, ko, common, reports, Achilles) {
 var a = new Achilles(...)

````

Something like that.  The point is that the Achilles component is loaded as a dependency and used during startup (assuming that's how it should work, I admit i haven't completely grok'd your code yet).

Also: full disclosure: AchillesWeb was one of the eraliest OHDSI apps we put together, and we learned a lot since then about AMD, knockout components, etc, so some of what I suggest may not fit in with the current architecture of AchillesWeb. But, I do like the direction that you are thinking, being able to define datasource types in the config, so I am very much interested in folding this in!